daffodil-core/src/main/scala/org/apache/daffodil/dsom/ElementBase.scala

```diff
@@ -748,15 +745,22 @@ trait ElementBase
   final lazy val isFixedLength = {
     (lengthKind =:= LengthKind.Explicit && lengthEv.isConstant) ||
       isImplicitLengthString
-    // TODO: there are lots of other cases where things are fixed length
+    // FIXME: there are lots of other cases where things are fixed length
     // e.g., implicit length hexBinary uses maxLength for length in bytes
     // e.g., implicit length fixed-precision binary numbers (byte, short, int, long and unsigned thereof)
     // In general the things in this file about fixed length seem to miss hexBinary.
+    // Also, things like packed and zoned decimal are usually fixed length, sometimes delimited.
   }
 
   final def isImplicitLengthString = isSimpleType && primType =:= PrimType.String && lengthKind =:= LengthKind.Implicit
 
   final lazy val fixedLengthValue: Long = {
+    // FIXME: this calculation only good for string type.
```

Comment:
Better is to deprecate this and change usages to use the maybeFixedLength method. That still needs to improve what it provides for answers to cover every type and representation.